api/src/main/java/io/opentelemetry/common/Attributes.java

```diff
@@ -155,12 +155,32 @@ public Attributes build() {
       return sortAndFilterToAttributes(data.toArray());
     }
 
+    private boolean checkKeyValue(String key, AttributeValue value) {
```

Comment:
How about `shouldAdd` and return false when it shouldn't? Having trouble following this method

Reply:
I would suggest calling it `isInvalidKeyValue`. This way, we don't need to negate its results in the `setAttribute` methods.

Reply:
doNotAdd could be ok too, though I don't know if I personally find it more readable than a negation. But the main point is to not have key in the method name since it's not only checking the key but doing something special based on the value as well. Just describing the return value (whether to add or not) seems fine to me.

Reply:
done in ed949be